packages/cli/src/commands/test.ts

```diff
@@ -164,45 +163,51 @@ export default class Test extends Command {
         hasFailures: true,
         runError: message,
       })
-      process.exitCode = 1
+      throw new Error(`Scheduler failure:, ${message}.`)
     })
     runner.on(Events.RUN_FINISHED, () => reporter.onEnd())
     await runner.run()
   }
 
-  // eslint-disable-next-line require-await
   async prepareRunLocation (
     configOptions: { runLocation?: string, privateRunLocation?: string } = {},
     cliFlags: { runLocation?: string, privateRunLocation?: string },
   ): Promise<RunLocation> {
     // Command line options take precedence
     if (cliFlags.runLocation) {
-      return { type: 'PUBLIC', region: cliFlags.runLocation }
+      const { data: availableLocations } = await api.locations.getAll()
+      if (availableLocations.some(l => l.region === cliFlags.runLocation)) {
+        return Promise.resolve({ type: 'PUBLIC', region: cliFlags.runLocation })
```

Comment:
Since this an `async` function, can we just return the value without calling `Promise.resolve`?